src/bootstrap/src/ferrocene/doc.rs

```diff
@@ -188,39 +192,36 @@ impl<P: Step + IsSphinxBook> Step for SphinxBook<P> {
             // Store doctrees outside the output directory:
             .arg("-d")
             .arg(relative_path(&src, &doctrees))
-            // Include the breadcrumbs
-            .arg("-D")
-            .arg(path_define(
-                "html_theme_options.include_in_header",
-                &relative_path(&src, &breadcrumbs.join("sphinx-template.html")),
-            ))
-            .arg("-A")
-            .arg(format!("ferrocene_breadcrumbs_index={path_to_root}/index.html"))
-            .arg("-D")
-            .arg(format!(
-                "rustfmt_version={}",
-                builder.crates.get("rustfmt-nightly").unwrap().version,
-            ))
-            // Include the CSS for the breadcrumbs. Note that the path here is relative to the
-            // _static directory in the rendered output. The directive works only because before
-            // invoking Sphinx we copy the CSS file into _static manually.
-            .arg("-D")
-            .arg("html_css_files=ferrocene-breadcrumbs.css")
             // Provide the correct substitutions:
-            .arg("-D")
             .arg(path_define("ferrocene_substitutions_path", &relative_path(&src, &substitutions)))
             // Provide the correct target names:
-            .arg("-D")
             .arg(path_define("ferrocene_target_names_path", &relative_path(&src, &target_names)))
             // Toolchain versions
-            .arg("-D")
-            .arg(format!("ferrocene_version={ferrocene_version}"))
-            .arg("-D")
+            .arg(format!("-Dferrocene_version={ferrocene_version}"))
             .arg(format!(
-                "rust_version={}",
+                "-Drust_version={}",
                 fs::read_to_string(&builder.src.join("src").join("version")).unwrap().trim(),
+            ))
+            .arg(format!(
+                "-Drustfmt_version={}",
+                builder.crates.get("rustfmt-nightly").unwrap().version,
             ));
 
+        // Include the public-docs warning message.
+        css_files.push(format!("{path_to_root}/../public-docs-warning.css"));
+        include_in_header.push(relative_path(&src, &public_docs_warning.join("header.html")));
```

Comment:
In my local copy (before copying the warning css over) I noticed this left a hanging request:
![image](https://github.com/user-attachments/assets/e3adcaf7-fa45-4674-91ae-0d0063b9e7a4)

I'm not able to come up with a good way to make this not happen, but perhaps by default we could have this be an empty file or something, so that the request doesn't just hang?

Reply:
This is not the first time we use this approach. We also rely on it to show/hide the technical report in the documentation index, and to show/hide the breadcrumbs in the documents. You can see the 404 happening in public-docs for `breadcrumbs.js` for example.